src/box/iproto.cc

```diff
@@ -1283,6 +1287,35 @@ iproto_connection_input_buffer(struct iproto_connection *con)
 	return new_ibuf;
 }
 
+static struct obuf *
+iproto_connection_output_buffer(struct iproto_connection *con)
+{
+	struct obuf_svp obuf_end = obuf_create_svp(con->wpos.obuf);
+	struct obuf_svp *begin = &con->wpos.svp;
+	if (con->wend.obuf != con->wpos.obuf) {
+		/*
+		 * Flush the current buffer before
+		 * advancing to the next one.
+		 */
+		if (begin->used == obuf_end.used) {
+			con->wpos.obuf = con->wend.obuf;
+			obuf_svp_reset(begin);
+		}
+	}
+	return con->wpos.obuf;
+}
+
+static inline bool
+iproto_flushed(struct iproto_connection *con)
```

Comment:
Do we really need this refactoring? Can't we just check that both output buffers are empty?

Reply:
Fixed.